.github/workflows/build.yml

```diff
@@ -99,29 +99,92 @@ jobs:
           rm -rf $(ls -1 --ignore=*.tgz --ignore=ci --ignore=t --ignore=utils --ignore=.github)
           tar zxvf ${{ steps.branch_env.outputs.fullname }}
 
-      - name: Start CI env (FIRST_TEST)
-        if: steps.test_env.outputs.type == 'first'
-        run: |
-          # launch deps env
-          make ci-env-up project_compose_ci=ci/pod/docker-compose.${{ steps.test_env.outputs.type }}.yml
+      - name: Cache images
+        id: cache-images
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-apisix-docker-images
+        with:
+          path: docker-images-backup
+          key: ${{ runner.os }}-${{ env.cache-name }}-${{ matrix.os_name }}-${{ steps.test_env.outputs.type }}-${{ hashFiles('./ci/pod/*.yml') }}
```

Comment:
What is cached are docker images from docker-compose-${type}.yml. So, why calculate the hash value of *.yml？

Reply:
So that, if any of the docker compose file changes, it is very likely that images might have been updated(version update)/added/removed. So changes to docker compose file invalidates the cache.

Reply:
What I want to say is that you should only calculate the hash value of `docker-compose-${type}.yml `, what is the result of `${{ hashFiles('./ci/pod/*.yml') }}` ?

Reply:
Currently, all docker images are being saved after extracting the image names from all docker compose files, with a key containing the combined hash of all docker compose files.

<img width="1128" alt="image" src="https://user-images.githubusercontent.com/61597896/215317232-a455e104-ee99-4746-82a5-55ade0afaebd.png">


So if we want to use hash value of  only `docker-compose-${type}.yml`, then I think we should only save images present in `docker-compose-${type}.yml` only, not from all the docker compse files. WDYT?

Reply:
You may not understand github action matrix: https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs
It has a "fission effect".
'first', 'plugin', and 'last' will be several independent CI tasks that do not affect each other

Reply:
> It has a "fission effect".
Oh yes! What I meant to say is using hash value of only `docker-compose-${type}.yml` will lead to multiple cache files containing the same docker images.

So if we change `${{ hashFiles('./ci/pod/*.yml') }}` to `${{ hashFiles('.ci/pod/docker-compose-${type}.yml') }}`. Then we should also update the part that is saving the docker images, i.e save images mentioned in the `.ci/pod/docker-compose-${type}.yml` file only (instead of saving images mentioned in all the docker compose files).

Reply:
My question is which path should we prefer? Saving all docker images mentioned in all docker compose files in a single cache file.

Or saving docker images mentioned in each docker compose file (first or last or plugin... ..) in a different cache file.

I think the second one sounds more efficient, but I am waiting for your opinion.

Reply:
The second method should be used, this method has as few cached docker images as possible. If we update the images in docker-compose.first, should the cache still be valid when executing plugin tasks? So the hash value only needs to be calculated for a certain docker compose file, not all